Model/Data/CollectData.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Alma\MonthlyPayments\Model\Data;
+
+use Alma\API\Entities\MerchantData\CmsFeatures;
+use Alma\API\Entities\MerchantData\CmsInfo;
+use Alma\API\Lib\PayloadFormatter;
+use Alma\API\Lib\RequestUtils;
+use Alma\MonthlyPayments\Api\Data\CollectDataInterface;
+use Alma\MonthlyPayments\Gateway\Config\Config;
+use Alma\MonthlyPayments\Helpers\ApiConfigHelper;
+use Alma\MonthlyPayments\Helpers\CmsFeaturesDataHelper;
+use Alma\MonthlyPayments\Helpers\CmsInfoDataHelper;
+use Alma\MonthlyPayments\Helpers\Logger;
+use Alma\MonthlyPayments\Model\Exceptions\CollectDataException;
+use Magento\Framework\HTTP\PhpEnvironment\Request;
+use Magento\Framework\Webapi\Exception;
+
+class CollectData implements CollectDataInterface
+{
+
+    private $payloadFormatter;
+    private $cmsInfoDataHelper;
+    private $cmsFeaturesDataHelper;
+    /**
+     * @var Request
+     */
+    private $request;
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * @var ApiConfigHelper
+     */
+    private $apiConfigHelper;
+
+
+    const HEADER_SIGNATURE_KEY = 'X-Alma-Signature';
+    /**
+     * @var Config
+     */
+    private $config;
+
+
+    /**
+     * @param Logger $logger
+     * @param PayloadFormatter $payloadFormatter
+     * @param CmsInfoDataHelper $cmsInfoDataHelper
+     * @param CmsFeaturesDataHelper $cmsFeaturesDataHelper
+     * @param Request $request
+     * @param ApiConfigHelper $apiConfigHelper
+     * @param Config $config
+     */
+    public function __construct(
+        Logger                $logger,
+        PayloadFormatter      $payloadFormatter,
+        CmsInfoDataHelper     $cmsInfoDataHelper,
+        CmsFeaturesDataHelper $cmsFeaturesDataHelper,
+        Request               $request,
+        ApiConfigHelper       $apiConfigHelper,
+        Config                $config
+    )
+    {
+        $this->payloadFormatter = $payloadFormatter;
+        $this->cmsInfoDataHelper = $cmsInfoDataHelper;
+        $this->cmsFeaturesDataHelper = $cmsFeaturesDataHelper;
+        $this->request = $request;
+        $this->logger = $logger;
+        $this->apiConfigHelper = $apiConfigHelper;
+        $this->config = $config;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect()
+    {
+        try {
+            $this->checkSignature($this->config->getMerchantId(), $this->getApiKey());
+        } catch (CollectDataException $e) {
+            throw new Exception(__($e->getMessage()), 0, Exception::HTTP_FORBIDDEN);
```

Comment:
throw Exception is it a good practice ? We don't need to throw a specific Exception from our site ?